smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/StructuredMemberWriter.java

```diff
@@ -123,32 +134,33 @@ private void writeCollectionFilterSensitiveLog(
         if (collectionMember.getMemberTrait(model, SensitiveTrait.class).isPresent()) {
             // member is Sensitive, hide the value.
             writer.write("SENSITIVE_STRING");
-            return;
+        } else if (model.expectShape(collectionMember.getTarget()) instanceof SimpleShape) {
+            writer.write(collectionParam);
+        } else {
+            writer.openBlock("$L.map(", ")", collectionParam, () -> {
+                String itemParam = "item";
+                Shape collectionMemberTarget = model.expectShape(collectionMember.getTarget());
+                writer.write("$L => ", itemParam);
+                if (collectionMemberTarget.isStructureShape() || collectionMemberTarget.isUnionShape()) {
```

Comment:
This looks really similar to the code in `writeMemberFilterSensitiveLog`, is it possible to refactor this chunk of `if/else` to instead use that? I might be missing something, but the final `else` is the only thing that looks different to me here.

Reply:
The purpose of the `else` block is to throw CodegenException in case redundant code is being written for provided models at Runtime for SimpleShape.

But let me verify that in a Unit test instead, and reuse code from `writeMemberFilterSensitiveLog`

Reply:
> But let me verify that in a Unit test instead, and reuse code from `writeMemberFilterSensitiveLog`

Code reused in commit 4d94416102db3175aa7d93cf2285927ccd639a8a
Unit test `test-insensitive-list` already checks for redundant code not being present.